MaterialLib/MPL/Phase.cpp

```diff
@@ -0,0 +1,134 @@
+/**
+ * \file
+ * \author Norbert Grunwald
+ * \date   Sep 7, 2017
+ *
+ * \copyright
+ * Copyright (c) 2012-2018, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#include "Phase.h"
+#include "BaseLib/Algorithm.h"
+#include "Properties/Properties.h"
+
+#include "Component.h"
+
+namespace MaterialPropertyLib
+{
+Phase::Phase(std::string const& phase_name)
+{
+    std::array<std::string, 4> allowed_phase_names = {
+        {"solid", "aqueous liquid", "non-aqueous liquid", "gas"}};
+
+    if (std::find(allowed_phase_names.begin(),
+                  allowed_phase_names.end(),
+                  phase_name) == allowed_phase_names.end())
+    {
+        ERR("Phase name should be one of:");
+        for (auto const name : allowed_phase_names)
+        {
+            ERR(name.c_str());
+        }
+        OGS_FATAL("Wrong phase name '%s' given.", phase_name.c_str());
+    }
+    _properties[name] = std::make_unique<Constant>(phase_name);
```

Comment:
The same question: can _properties be initialized in the initialization list of this ctor?